src/components/pages/account/InvestmentChart.tsx

```diff
@@ -50,61 +52,161 @@ export default function InvestmentChart({
   const numStocksData: { x: number, y: number }[] = [];
   const valueData: { x: number, y: number }[] = [];
   const dates = intervalToDates(interval);
+  const isCurrencyInvestment = investment.account.commodity.namespace === 'CURRENCY';
 
-  dates.forEach(x => {
-    investment.processSplits(x);
-    numStocksData.push({
-      x: x.startOf('month').toMillis(),
-      y: investment.quantity.toNumber(),
+  investment.splits
+    .filter(s => interval.contains(s.transaction.date) && !InvestmentAccount.isDividend(s))
+    .forEach(s => {
+      investment.processSplits(s.transaction.date);
+      numStocksData.push({
+        x: s.transaction.date.toMillis(),
+        y: !isCurrencyInvestment ? investment.quantity.toNumber() : investment.cost.toNumber(),
+      });
+
+      const price = pricesMap.getInvestmentPrice(
+        investment.account.commodity.mnemonic,
+        s.transaction.date,
+      );
+
+      if (price) {
+        valueData.push({
+          x: s.transaction.date.toMillis(),
+          y: investment.quantity.toNumber() * price.value,
+        });
+      }
     });
 
-    const price = findClosestPrice(x, prices || []);
-    if (price) {
-      valueData.push({
+  dates.forEach(x => {
+    const alreadyExists = numStocksData.find(d => d.x === x.toMillis());
+
+    if (!alreadyExists) {
+      investment.processSplits(x);
+      numStocksData.push({
         x: x.toMillis(),
-        y: investment.quantity.toNumber() * price.value,
+        y: !isCurrencyInvestment ? investment.quantity.toNumber() : investment.cost.toNumber(),
```

Comment:
when it is in currency, we want to display the cost as "quantity" will show the value (because there is no alternative commodity)